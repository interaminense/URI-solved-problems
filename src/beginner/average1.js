/**
 * @author Adriano Interaminense <adriano.interaminense@gmail.com>
 *
 * Read two floating points' values of double precision A and B, corresponding
 * to two student's grades. After this, calculate the student's average,
 * considering that grade A has weight 3.5 and B has weight 7.5. Each grade can
 * be from zero to ten, always with one digit after the decimal point.
 *
 * INPUT
 *
 * The input file contains 2 floating points values with one digit after the
 * decimal point.
 * @param {number} A
 * @param {number} B
 *
 * OUTPUT
 *
 * Print AVERAGE according to the following example, with 5 digits after the
 * decimal point and with a blank space before and after the equal signal.
 */

const tmpl = result => `AVERAGE = ${result}`;

const average = (A, B) => {
  const result = ((A * 3.5) + (B * 7.5)) / (3.5 + 7.5);

  return tmpl(result.toFixed(5));
};

export { average, tmpl };
