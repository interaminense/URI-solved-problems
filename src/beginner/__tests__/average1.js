import { average, tmpl } from '../average1';

describe('Average 1', () => {
  let A = 0;
  let B = 0;

  it('The average for 5.0 and 7.1 must be 6.43182', () => {
    A = 5.0;
    B = 7.1;

    expect(average(A, B)).toEqual(tmpl((6.43182).toFixed(5)));
  });

  it('The average for 0.0 and 7.1 must be 4.84091', () => {
    A = 0.0;
    B = 7.1;

    expect(average(A, B)).toEqual(tmpl((4.84091).toFixed(5)));
  });

  it('The average 10.0 + 10.0 must be 10.0', () => {
    A = 10.0;
    B = 10.0;

    expect(average(A, B)).toEqual(tmpl((10.0).toFixed(5)));
  });
});
