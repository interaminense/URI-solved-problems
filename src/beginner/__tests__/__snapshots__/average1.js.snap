// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Average 1 The average 10.0 + 10.0 must be 10.0 1`] = `"AVERAGE = 10.00000"`;

exports[`Average 1 The average for 0.0 and 7.1 must be 4.84091 1`] = `"AVERAGE = 4.84091"`;

exports[`Average 1 The average for 5.0 and 7.1 must be 6.43182 1`] = `"AVERAGE = 6.43182"`;
